### PR TITLE
Using specific role instead of cluster admin to deploy registration agent

### PR DESCRIPTION
**Pull request by skeeey**
Using specific role instead of cluster admin to deploy registration agent, so make sure the e2e test more credible

### CODE REVIEW

File: deploy/spoke/role.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: open-cluster-management:spoke
```
Comment (pmorie) [inline deploy/spoke/role.yaml]: Let's think carefully about using the name `spoke` here. `agent` might be a better alternative. Do you have a preference on this detail, @mdelder ?

**Review by pmorie — COMMENTED** [review]
Good change, let's just make sure we have the right name

File: deploy/spoke/role.yaml (hunk 1 above)
Comment (skeeey) [reply deploy/spoke/role.yaml]: how about `registration-agent`? I think this more specific

**Comment by mdelder** [comment]
I agree with @pmorie on this one. I would be vote for either registration-agent or klusterlet as the name for the role.

**Comment by skeeey** [comment]
using `registration-agent` first

File: deploy/spoke/clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:spoke
```
Comment (deads2k) [inline deploy/spoke/clusterrole.yaml:4]: I think we re-branded "managed-cluster" so you may want to update this.

**Pull request merged**